### DISCUSSION TITLE
I'm trying to deploy a standalone Next.js app on Hostinger shared hosting, but I'm encountering a 403 error

### DISCUSSION THREAD

**Discussion by itsZain1** [vercel/next.js#72805] [Help]
### Summary

I'm exporting a standalone Next.js project and deploying it on Hostinger shared hosting, but I'm getting a 403 Forbidden error. The app is static with no API calls.

My next.config.js:
`const config = require("./src/config/config.json");
const languages = require("./src/config/language.json");
const disableLanguages = config.settings.disable_languages;
const activeLanguages = languages.filter(
  (lang) => !disableLanguages.includes(lang.languageCode),
);

const defaultLanguage = config.settings.default_language;

const otherLanguages = activeLanguages
  .map((lang) => lang.languageCode)
  .filter((lang) => lang !== defaultLanguage);

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  basePath: config.base_path !== "/" ? config.base_path : "",
  trailingSlash: config.site.trailing_slash,
  output: "standalone",
  images: {
    domains: ["res.cloudinary.com"],
  },
  async rewrites() {
    if (config.settings.default_language_in_subdir) {
      return [];
    }

    return activeLanguages.length !== 1
      ? [
          {
            source: `/:lang(!${defaultLanguage}|${otherLanguages.join("|")})/:path*`,
            destination: `/:lang/:path*`,
          },
          {
            source: `/:path*`,
            destination: `/${defaultLanguage}/:path*`,
          },
        ]
      : [
          {
            source: `/${defaultLanguage}/:path*`,
            destination: `/${defaultLanguage}/:path*`,
          },
          {
            source: "/:path*",
            destination: `/${defaultLanguage}/:path*`,
          },
        ];
  },
};

module.exports = nextConfig;`

Locally, the .next build is generated successfully, and I'm using output: 'standalone'. When I tried using output: 'export', I encountered an error.

<img width="1179" alt="Screenshot 2024-11-14 at 2 58 22 PM" src="https://github.com/user-attachments/assets/c3a51891-6c6c-40bf-b27f-851ac17ce35f">

And here's what my hostinger file manager looks like:

![hpanel-filemanager](https://github.com/user-attachments/assets/320fdadb-e948-4eb2-b15c-726448cd96bc)

Any help would be appreciated!

**Comment by icyJoseph** [comment]
Does your Hostinger setup, support Node.js?

standalone requires a Node.js server to run your application...

I see here, https://support.hostinger.com/en/articles/1583661-is-node-js-supported-at-hostinger, that Hostinger, does have Node.js, but it looks to me as if you want to have a static HTML site, which is why you mention `export` mode as well.

If you just want to do static hosting, you need to use `export` mode yes.

And that has implication, because not all features are supported.

The error you got when using `export` is because, you need to provide upfront the paths you'd want to generate when `export`ing to a static HTML site.

Could you confirm that you want a static HTMl site? Before I write way more...
